dsc_lib/src/functions/min.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::DscError;
+use crate::configure::context::Context;
+use crate::functions::{AcceptedArgKind, Function};
+use serde_json::Value;
+use tracing::debug;
+
+#[derive(Debug, Default)]
+pub struct Min {}
+
+impl Function for Min {
+    fn min_args(&self) -> usize {
+        1
+    }
+
+    fn max_args(&self) -> usize {
+        usize::MAX
+    }
+
+    fn accepted_arg_types(&self) -> Vec<AcceptedArgKind> {
+        vec![AcceptedArgKind::Number, AcceptedArgKind::Array]
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        debug!("min function");
+        if args.len() == 1 {
+            if let Some(array) = args[0].as_array() {
+                if array.len() > 1 {
+                    find_min(array)
+                }
+                else {
+                    Err(DscError::Parser("Array must contain more than 1 integer".to_string()))
```

Comment:
Is this true or would it just return the one integer?

Reply:
I based this on the description within the documentation since it only refers to "integers" but yes, the code would just return the one integer